### ISSUE TITLE
4.7 IPI Install on OpenStack fails

### ISSUE THREAD

**Issue by avisiedo** [openshift/installer#4663]
# Version

```console
$ openshift-install version
openshift-install 4.7.0-0.nightly-2021-02-13-071408
built from commit 98e11541c24e95c864328b9b35c64b77836212ed
release image registry.ci.openshift.org/ocp/release@sha256:b18d4a98d8f9103694c9842a141f5a89a8b44e5675c878caaaf5174206d0f36d
```

# Platform:
openstack

IPI

# What happened?

I got the following output:

```raw
level=debug msg=Apply complete! Resources: 62 added, 0 changed, 0 destroyed.
level=debug msg=OpenShift Installer 4.7.0-0.nightly-2021-02-13-071408
level=debug msg=Built from commit 98e11541c24e95c864328b9b35c64b77836212ed
level=info msg=Waiting up to 20m0s for the Kubernetes API at https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443...
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.135.174:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.135.174:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.135.174:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.135.174:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
level=error msg=Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.0.135.174:6443: connect: no route to host
level=debug msg=Fetching Bootstrap SSH Key Pair...
level=debug msg=Loading Bootstrap SSH Key Pair...
level=debug msg=Using Bootstrap SSH Key Pair loaded from state file
level=debug msg=Reusing previously-fetched Bootstrap SSH Key Pair
level=debug msg=Fetching Install Config...
level=debug msg=Loading Install Config...
level=debug msg=  Loading SSH Key...
level=debug msg=  Loading Base Domain...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Cluster Name...
level=debug msg=    Loading Base Domain...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Networking...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Pull Secret...
level=debug msg=  Loading Platform...
level=debug msg=Using Install Config loaded from state file
level=debug msg=Reusing previously-fetched Install Config
level=error msg=failed to lookup masters: resource not found
level=info msg=Pulling debug logs from the bootstrap machine
level=debug msg=Using SSH_AUTH_SOCK /tmp/ssh-KlyMrG8AWwgC/agent.24 to connect to an existing agent
level=debug msg=Gathering bootstrap systemd summary ...
level=debug msg=Gathering bootstrap failed systemd unit status ...
level=debug msg=Gathering bootstrap journals ...
level=debug msg=Gathering bootstrap containers ...
level=debug msg=time="2021-02-16T22:02:43Z" level=fatal msg="connect: connect endpoint 'unix:///var/run/crio/crio.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded"
level=debug msg=cp: cannot stat '/var/log/bootstrap-control-plane/': No such file or directory
level=debug msg=Gathering rendered assets...
level=debug msg=Gathering cluster resources ...
level=debug msg=Waiting for logs ...
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: dial tcp 10.0.135.174:6443: i/o timeout
level=debug msg=Unable to connect to the server: dial tcp 10.0.135.174:6443: i/o timeout
level=debug msg=Unable to connect to the server: dial tcp 10.0.135.174:6443: i/o timeout
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: context deadline exceeded (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: dial tcp 10.0.133.58:6443: i/o timeout
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
level=debug msg=Unable to connect to the server: dial tcp 10.0.135.174:6443: i/o timeout (Client.Timeout exceeded while awaiting headers)
level=debug msg=Gather remote logs
level=debug msg=No masters found!
level=debug msg=Log bundle written to /var/home/core/log-bundle-20210216220242.tar.gz
level=info msg=Bootstrap gather logs captured here "/builds/avisiedo/idm-ocp-provisioner/clusters/dp-031ccfef/log-bundle-20210216220242.tar.gz"
level=fatal msg=Bootstrap failed to complete: failed waiting for Kubernetes API: Get "https://api.dp-031ccfef.idmocp.lab.eng.rdu2.redhat.com:6443/version?timeout=32s": dial tcp 10.0.133.58:6443: connect: no route to host
```

I observe in OpenStack that boostrap and master plane vms are created, but nothing about the worker nodes.

# What you expected to happen?

```raw
level=info msg=Waiting up to 20m0s for the Kubernetes API at https://api.dp-b1fe87d0.idmocp.lab.eng.rdu2.redhat.com:6443...
level=info msg=API v1.20.0+d9c52cc up
level=info msg=Waiting up to 30m0s for bootstrapping to complete...
level=debug msg=Bootstrap status: complete
level=info msg=Destroying the bootstrap resources...

. . .

level=info msg=Waiting up to 40m0s for the cluster at https://api.dp-b1fe87d0.idmocp.lab.eng.rdu2.redhat.com:6443 to initialize...
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 80% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: downloading update
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 0% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 7% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 11% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 62% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 88% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 95% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 96% complete
level=debug msg=Still waiting for the cluster to initialize: Cluster operator authentication is reporting a failure: WellKnownReadyControllerDegraded: kube-apiserver oauth endpoint https://10.8.2.191:6443/.well-known/oauth-authorization-server is not yet served and authentication operator keeps waiting (check kube-apiserver operator, and check that instances roll out successfully, which can take several minutes per instance)
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 98% complete
level=debug msg=Still waiting for the cluster to initialize: Working towards 4.7.0-0.nightly-2021-01-19-095812: 99% complete
level=debug msg=Still waiting for the cluster to initialize: Cluster operator authentication is reporting a failure: WellKnownReadyControllerDegraded: kube-apiserver oauth endpoint https://10.8.2.191:6443/.well-known/oauth-authorization-server is not yet served and authentication operator keeps waiting (check kube-apiserver operator, and check that instances roll out successfully, which can take several minutes per instance)
level=debug msg=Still waiting for the cluster to initialize: Cluster operator authentication is reporting a failure: WellKnownReadyControllerDegraded: need at least 3 kube-apiservers, got 2
level=debug msg=Cluster is initialized
level=info msg=Waiting up to 10m0s for the openshift-console route to be created...
level=debug msg=Route found in openshift-console namespace: console
level=debug msg=OpenShift console route is admitted
level=info msg=Install complete!
level=info msg=To access the cluster as the system:admin user when using 'oc', run 'export KUBECONFIG=/builds/avisiedo/idm-ocp-provisioner/clusters/dp-b1fe87d0/auth/kubeconfig'
level=info msg=Access the OpenShift web-console here: https://console-openshift-console.apps.dp-b1fe87d0.idmocp.lab.eng.rdu2.redhat.com
level=info msg=Login to the console with user: "*********", and password: "*********************"
level=debug msg=Time elapsed per stage:
level=debug msg=    Infrastructure: 4m20s
level=debug msg=Bootstrap Complete: 9m7s
level=debug msg= Bootstrap Destroy: 39s
level=debug msg= Cluster Operators: 12m32s
level=info msg=Time elapsed: 28m6s

```

# How to reproduce it (as minimally and precisely as possible)?

I am launching openshift-install from inside a container which contain all the needed dependencies, in IPI mode. It basically generate the configuration file, and eventually is launching:

```console
$ openshift-install --dir=clusters/my-cluster-id --log-level debug create cluster
```

# Anything else we need to know?

The last detected version which worked is **4.7.0-0.nightly-2021-01-19-095812**. We are launching the last one in a pipeline at night for detecting which version works for investigation purpose for OpenShift 4.7

**Comment by staebler** [comment]
Check in log-bundle-20210216220242.tar.gz under bootstrap/journals. You will likely see an error in either release-image.log or bootkube.log.

**Comment by avisiedo** [comment]
Hi @staebler 

I have just seen at release-image.log the following:

```raw
Feb 16 21:40:32 dp-031ccfef-9jpzk-bootstrap systemd[1]: Starting Download the OpenShift Release Image...
Feb 16 21:40:32 dp-031ccfef-9jpzk-bootstrap release-image-download.sh[1709]: Pulling registry.ci.openshift.org/ocp/release@sha256:b18d4a98d8f9103694c9842a141f5a89a8b44e5675c878caaaf5174206d0f36d...
Feb 16 21:40:34 dp-031ccfef-9jpzk-bootstrap release-image-download.sh[1709]: Error: unable to pull registry.ci.openshift.org/ocp/release@sha256:b18d4a98d8f9103694c9842a141f5a89a8b44e5675c878caaaf5174206d0f36d: Error initializing source docker://registry.ci.openshift.org/ocp/release@sha256:b18d4a98d8f9103694c9842a141f5a89a8b44e5675c878caaaf5174206d0f36d: Error reading manifest sha256:b18d4a98d8f9103694c9842a141f5a89a8b44e5675c878caaaf5174206d0f36d in registry.ci.openshift.org/ocp/release: unauthorized: authentication required
```

By that message it seems my pull-secret is not correct. Currently I am using artifacts from [this place](https://openshift-release-artifacts.apps.ci.l2s4.p1.openshiftapps.com/). By the way, I have seen [this other place](https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/), which probably are the binaries I used into the container that is working. I have launched the provision process collecting that binaries and everything worked fine.

In short: If I want to use the latest CI, I need to update my pull-secrets used into my pipeline; if the version from the mirror is enough, I have just to change the references I am using for collecting the openshift-installer and openshift-clients to that mirror which is working without additional pull-secrets, when building the container ready for provisioning OpenShift 4.7 clusters.

Thanks a million for pointing me out to the right place :+1:
